system/default_ad_api/src/routing.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROUTING_HPP_
+#define ROUTING_HPP_
+
+#include "utils/types.hpp"
+
+#include <autoware_ad_api_specs/routing.hpp>
+#include <component_interface_specs/planning.hpp>
+#include <component_interface_utils/rclcpp.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+namespace default_ad_api
+{
+
+class RoutingNode : public rclcpp::Node
+{
+public:
+  explicit RoutingNode(const rclcpp::NodeOptions & options);
+
+private:
+  USING_COMPONENT_INTERFACE_WRAPPER_ALIAS
+  rclcpp::CallbackGroup::SharedPtr group_srv_;
+  Pub<autoware_ad_api::routing::RouteState> pub_route_state_;
```

Comment:
[IMO] I personally feel avoiding the macro and using the following way is more readable for wide developers.

```cpp
using component_interface_utils::Publisher;
Publisher<autoware_ad_api::routing::RouteState>::SharedPtr pub_route_state_;
```

Reply:
It cannot work. For template, need the following. Writing 4 types of this will be 8 lines, so I made it a macro.
```cpp
template <class T>
using Publisher = component_interface_utils::Publisher<T>;
```

Reply:
Mm, is that true?
https://wandbox.org/permlink/FMyjsBxtTLSwvxvC

Reply:
It cannot be used in class scope.
https://wandbox.org/permlink/QTGDi8L5ZRHKRVjf

Reply:
I see. In that case, why not do it outside the class scope (= inside the namespace)?
I believe macros should be avoided unless we really need them.

Reply:
Done https://github.com/autowarefoundation/autoware.universe/pull/1494/commits/962c2e9a0a7c01e340b72b910b2be7ac1497528e.

Reply:
Fix build error 8f0cba12316c1045400d9c70d0950be3baebc5fe.